### ISSUE TITLE
Add metric for every step on the pipeline

### ISSUE THREAD

**Issue by kaisbaccour** [noislabs/pingpong#2]
- beacons_request_inclsuion_in_customer_chain_time
- relay_and_job_subscription_time
- drand_wait_time
- verification_time
- relay_and_block_inclsuion_in_customer_chain_time

**Comment by webmaster128** [comment]
This means many different histograms, right? So far we have just two, the e2e time and the processing time (i.e. the part we can control).

**Comment by kaisbaccour** [comment]
> This means many different histograms, right? So far we have just two, the e2e time and the processing time (i.e. the part we can control).

Yes indeed so probably 5 histograms. Then the timing out task  would get the max_integer.
We could eventually add the ics20 transfer time but that's probably not as important.

**Comment by webmaster128** [comment]
When we track all of this data, we need to ensure to not stack the histograms. Some of them track overlapping times such as relaying to the gateway and waiting for the drand.
